aptos-move/framework/src/natives/cryptography/groups/arkworks_backend.rs

```diff
@@ -0,0 +1,1896 @@
+    pub fr_mul: InternalGas,
+    pub fr_neg: InternalGas,
+    pub fr_rand: InternalGas,
+    pub fr_serialize: InternalGas,
+    pub fr_sub: InternalGas,
+    pub fr_to_repr: InternalGasPerArg,
+    pub fq12_clone: InternalGas,
+    pub fq12_deserialize: InternalGasPerArg,
+    pub fq12_eq: InternalGasPerArg,
+    pub fq12_inv: InternalGas,
+    pub fq12_mul: InternalGasPerArg,
+    pub fq12_one: InternalGas,
+    pub fq12_pow_fr: InternalGasPerArg,
+    pub fq12_serialize: InternalGas,
+    pub fq12_square: InternalGas,
+    pub g1_affine_add: InternalGas,
+    pub g1_affine_deserialize_compressed: InternalGas,
+    pub g1_affine_deserialize_uncompressed: InternalGas,
+    pub g1_affine_eq_proj: InternalGas,
+    pub g1_affine_generator: InternalGas,
+    pub g1_affine_infinity: InternalGas,
+    pub g1_affine_mul_to_proj: InternalGas,
+    pub g1_affine_neg: InternalGas,
+    pub g1_affine_serialize_uncompressed: InternalGas,
+    pub g1_affine_serialize_compressed: InternalGas,
+    pub g1_affine_to_prepared: InternalGasPerArg,
+    pub g1_affine_to_proj: InternalGas,
+    pub g1_proj_add: InternalGasPerArg,
+    pub g1_proj_addassign: InternalGasPerArg,
+    pub g1_proj_double: InternalGas,
+    pub g1_proj_eq: InternalGas,
+    pub g1_proj_generator: InternalGas,
+    pub g1_proj_infinity: InternalGas,
+    pub g1_proj_mul: InternalGasPerArg,
+    pub g1_proj_mulassign: InternalGasPerArg,
+    pub g1_proj_neg: InternalGas,
+    pub g1_proj_rand: InternalGas,
+    pub g1_proj_sub: InternalGas,
+    pub g1_proj_subassign: InternalGas,
+    pub g1_proj_to_affine: InternalGas,
+    pub g1_proj_to_prepared: InternalGasPerArg,
+    pub g2_affine_add: InternalGas,
+    pub g2_affine_deserialize_compressed: InternalGas,
+    pub g2_affine_deserialize_uncompressed: InternalGas,
+    pub g2_affine_eq_proj: InternalGas,
+    pub g2_affine_generator: InternalGas,
+    pub g2_affine_infinity: InternalGas,
+    pub g2_affine_mul_to_proj: InternalGas,
+    pub g2_affine_neg: InternalGas,
+    pub g2_affine_serialize_compressed: InternalGas,
+    pub g2_affine_serialize_uncompressed: InternalGas,
+    pub g2_affine_to_prepared: InternalGasPerArg,
+    pub g2_affine_to_proj: InternalGas,
+    pub g2_proj_add: InternalGasPerArg,
+    pub g2_proj_addassign: InternalGasPerArg,
+    pub g2_proj_double: InternalGas,
+    pub g2_proj_eq: InternalGas,
+    pub g2_proj_generator: InternalGas,
+    pub g2_proj_infinity: InternalGas,
+    pub g2_proj_mul: InternalGasPerArg,
+    pub g2_proj_mulassign: InternalGasPerArg,
+    pub g2_proj_neg: InternalGas,
+    pub g2_proj_rand: InternalGas,
+    pub g2_proj_sub: InternalGas,
+    pub g2_proj_subassign: InternalGas,
+    pub g2_proj_to_affine: InternalGas,
+    pub g2_proj_to_prepared: InternalGasPerArg,
+    pub pairing_product_base: InternalGas,
+    pub pairing_product_per_pair: InternalGasPerArg,
+}
+
+#[derive(Debug, Clone)]
+pub struct GasParameters {
+    pub bls12_381: Bls12381GasParameters,
```

Comment:
Why do you have this layer of indirection? Could've declared everything here directly, no?

Reply:
Fixed.